third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp

```diff
@@ -475,11 +475,29 @@ class BlockedToMFMA : public OpRewritePattern<tt::DotOp> {
     // requires to broadcast the operand A.
     bool isTransposed = !(mDim == 4 && nDim == 64);
     auto aElemTy = mfmaInstr->aElementType;
-    ttg::AMDMfmaEncodingAttr mfmaEnc = ttg::AMDMfmaEncodingAttr::get(
-        oldRetType.getContext(),
-        /*version*/ mfmaVersion, warpsPerTile,
-        /*instrShape*/ mDim, nDim, /*isTransposed=*/isTransposed, CTALayout,
-        mfmaAccType);
+    bool isMfma16InBwdFA =
```

Comment:
This check is both too broad and too rigid--e.g., for the former, we aren't checking this is FA bwd? For the latter, do we need to check N is exactly 16? Also we can move forward some utilities used in L545?

Reply:
The check hasTransInDefChain() is kind of rigid. For example, the 1st pair of dot op in hstu attn bwd don't have the required transpose op in their def chain and so they would fail the check. I also feel this check is too broad. So I just limited it to match the exactly dimensions in hstu attn bwd.